core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -1094,7 +1094,7 @@ public void setReplication(RpcContext rpcContext, LockedInodePath inodePath,
    * @return the set of file ids which are pinned
    */
   public Set<Long> getPinIdSet() {
-    return new HashSet<>(mState.getPinnedInodeFileIds());
+    return mState.getPinnedInodeFileIds();
```

Comment:
This `mState.getPinnedInodeFileIds()` creates a `Collections.unmodifieableSet()` over the set of pinned file IDs. 
https://github.com/Alluxio/alluxio/blob/4c4ec92c9a117dc5cebd668a0d3e5803c645e87e/core/server/master/src/main/java/alluxio/master/file/meta/InodeTreePersistentState.java#L133

I think we don't need to create a new HashSet over the UnmodifiableSet here. The only upside of having a `HashSet` is being modifiable. The downside is a copy of a whole set. Currently all the references of this method are read-only. This copy is unnecessary.